### PR TITLE
[in-place][needs-docs] add buffer for polygons and fix #20147 in place difference 

### PR DESCRIPTION
**Pull request by elpaso**
plus some unrelated small fixes:
- dox in QgsFeatureReques:  FetchAttributes does not exist 
- for some reason that I could not understand, there is a difference in calling:
```
def f(settings=QgsSettings()):
    print(settings.fileName())

# and 
def f(settings=None):
    if not settings:
        settings = QgsSettings()
    print(settings.fileName())

```

The first return the wrong (basically default) path to settings, while the second call prints the correct one (the user profile one), this was stepping on my toes while testing.

### CODE REVIEW

**Comment by DelazJ** [comment]
Thanks @elpaso. Could you please add a [needs-docs] tag so that we document buffer as a [compatible in-place edit algorithm](https://github.com/qgis/QGIS-Documentation/pull/3074)?

**Comment by luipir** [comment]
@DelazJ is there a reason why do not we use labels instead of [need-docs] in PR title?

**Comment by DelazJ** [comment]
@luipir Because labels are not saved in the commit when merging, and commit message is what [generates issue report](https://github.com/qgis/QGIS-Sysadmin/blob/master/webhooks/github_feature_tracker.cgi#L429) in doc repo.

File: python/plugins/processing/gui/AlgorithmExecutor.py [hunk 1]
```diff
@@ -178,7 +183,7 @@ def execute_in_place_run(alg, parameters, context=None, feedback=None, raise_exc
                 else:
                     active_layer.deleteFeature(f.id())
                     # Get the new ids
-                    old_ids = set([f.id() for f in active_layer.getFeatures(req)])
+                    old_ids = set([f.id() for f in active_layer.getFeatures()])
```
Comment (nyalldawson) [inline python/plugins/processing/gui/AlgorithmExecutor.py]: This change looks potentially very expensive -- can it be tightened at all?

File: src/analysis/processing/qgsoverlayutils.cpp [hunk 2]
```diff
@@ -92,6 +92,7 @@ void QgsOverlayUtils::difference( const QgsFeatureSource &sourceA, const QgsFeat
 
   QgsFeature featA;
   QgsFeatureRequest requestA;
+  requestA.setInvalidGeometryCheck( context.invalidGeometryCheck() );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsoverlayutils.cpp]: good catch

File: python/plugins/processing/gui/AlgorithmExecutor.py (hunk 1 above)
Comment (elpaso) [reply python/plugins/processing/gui/AlgorithmExecutor.py]: Yes: I wanted to add 'req' back but ran out of time yesterday night. I will fix it later today.

**Pull request merged**